Tests/UnitTests/Fatras/Physics/PhotonConversionTests.cpp

```diff
@@ -0,0 +1,173 @@
+  std::vector<ActsFatras::Particle> generated;
+  BOOST_CHECK(!pc.run(gen, particle, generated));
+  BOOST_CHECK(generated.empty());
+  // Particle shouldn't be modified
+  BOOST_CHECK_EQUAL(particle.fourPosition(), particleInit.fourPosition());
+  BOOST_CHECK_EQUAL(particle.fourMomentum(), particleInit.fourMomentum());
+  BOOST_CHECK_EQUAL(particle.process(), particleInit.process());
+  BOOST_CHECK_EQUAL(particle.properTime(), particleInit.properTime());
+  BOOST_CHECK_EQUAL(particle.pathInX0(), particleInit.pathInX0());
+  BOOST_CHECK_EQUAL(particle.pathInL0(), particleInit.pathInL0());
+}
+
+BOOST_DATA_TEST_CASE(LowMomentumPhoton, Dataset::parametersPhotonConversion,
+                     phi, lambda, seed) {
+  using Scalar = ActsFatras::PhotonConversion::Scalar;
+  using namespace Acts::UnitLiterals;
+
+  Generator gen(seed);
+
+  /// Produce a low momentum photon
+  ActsFatras::Particle particle =
+      Dataset::makeParticle(Acts::PdgParticle::eGamma, phi, lambda, 1_keV);
+  ActsFatras::Particle particleInit = particle;
+
+  ActsFatras::PhotonConversion pc;
+
+  // No limits should be set - momentum too low
+  std::pair<Scalar, Scalar> limits = pc.generatePathLimits(gen, particle);
+  BOOST_CHECK_EQUAL(limits.first, std::numeric_limits<Scalar>::infinity());
+  BOOST_CHECK_EQUAL(limits.second, std::numeric_limits<Scalar>::infinity());
+
+  // No particles should be generated - momentum too low
+  std::vector<ActsFatras::Particle> generated;
+  BOOST_CHECK(!pc.run(gen, particle, generated));
+  BOOST_CHECK(generated.empty());
+  // Particle shouldn't be modified
+  BOOST_CHECK_EQUAL(particle.fourPosition(), particleInit.fourPosition());
+  BOOST_CHECK_EQUAL(particle.fourMomentum(), particleInit.fourMomentum());
+  BOOST_CHECK_EQUAL(particle.process(), particleInit.process());
+  BOOST_CHECK_EQUAL(particle.properTime(), particleInit.properTime());
+  BOOST_CHECK_EQUAL(particle.pathInX0(), particleInit.pathInX0());
+  BOOST_CHECK_EQUAL(particle.pathInL0(), particleInit.pathInL0());
+}
+
+BOOST_DATA_TEST_CASE(HighMomentumPhoton, Dataset::parametersPhotonConversion,
+                     phi, lambda, seed) {
+  using Scalar = ActsFatras::PhotonConversion::Scalar;
+  using namespace Acts::UnitLiterals;
+
+  Generator gen(seed);
+
+  /// Produce a high momentum photon
+  ActsFatras::Particle particle =
+      Dataset::makeParticle(Acts::PdgParticle::eGamma, phi, lambda, 1_GeV);
+  ActsFatras::Particle particleInit = particle;
+
+  ActsFatras::PhotonConversion pc;
+
+  // No limits should be set - momentum too low
+  std::pair<Scalar, Scalar> limits = pc.generatePathLimits(gen, particle);
+  BOOST_CHECK_NE(limits.first, std::numeric_limits<Scalar>::infinity());
+  BOOST_CHECK_EQUAL(limits.second, std::numeric_limits<Scalar>::infinity());
+
+  // No particles should be generated - momentum too low
+  std::vector<ActsFatras::Particle> generated;
+  BOOST_CHECK(pc.run(gen, particle, generated));
+  BOOST_CHECK_EQUAL(generated.size(), 2);
```

Comment:
The documentation is wrong again. Rebase issues?

Reply:
This is why I usually don't commit suggestions... will fix it

Reply:
Done